gtfs-validator/main/src/test/java/org/mobilitydata/gtfsvalidator/validator/StopTooFarFromTripShapeValidatorTest.java

```diff
@@ -0,0 +1,164 @@
+/*
+ * Copyright 2020 Google LLC, MobilityData IO
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.mobilitydata.gtfsvalidator.validator;
+
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+import com.google.common.base.Preconditions;
+import java.util.ArrayList;
+import org.mobilitydata.gtfsvalidator.table.GtfsStop;
+import org.mobilitydata.gtfsvalidator.table.GtfsStopTableContainer;
+import org.mobilitydata.gtfsvalidator.table.GtfsLocationType;
+import org.mobilitydata.gtfsvalidator.table.GtfsStopTime;
+import org.mobilitydata.gtfsvalidator.table.GtfsStopTimeTableContainer;
+import org.mobilitydata.gtfsvalidator.table.GtfsShape;
+import org.mobilitydata.gtfsvalidator.table.GtfsShapeTableContainer;
+import org.mobilitydata.gtfsvalidator.table.GtfsTrip;
+import org.mobilitydata.gtfsvalidator.table.GtfsTripTableContainer;
+import org.mobilitydata.gtfsvalidator.notice.NoticeContainer;
+import org.mobilitydata.gtfsvalidator.notice.StopTooFarFromTripShapeNotice;
+
+import static com.google.common.truth.Truth.assertThat;
+
+@RunWith(JUnit4.class)
+public class StopTooFarFromTripShapeValidatorTest {
+
+    public static GtfsStop createStop(long csvRowNumber, String stopId, double stopLat, double stopLon, GtfsLocationType locationType) {
+        return new GtfsStop.Builder().setCsvRowNumber(csvRowNumber).setStopId(stopId).setStopLat(stopLat).setStopLon(stopLon).setLocationType(locationType.getNumber()).build();
+    }
+
+    public static GtfsStopTime createStopTime(long csvRowNumber, String tripId, String stopId, int stopSequence) {
+        return new GtfsStopTime.Builder().setCsvRowNumber(csvRowNumber).setTripId(tripId).setStopId(stopId).setStopSequence(stopSequence).build();
+    }
+
+    public static GtfsShape createShape(long csvRowNumber, String shapeId, double shapePtLat, double shapePtLon, int shapePtSequence) {
+        return new GtfsShape.Builder().setCsvRowNumber(csvRowNumber).setShapeId(shapeId).setShapePtLat(shapePtLat).setShapePtLon(shapePtLon).setShapePtSequence(shapePtSequence).build();
+    }
+
+    public static GtfsTrip createTrip(long csvRowNumber, String tripId, String shapeId) {
+        return new GtfsTrip.Builder().setCsvRowNumber(csvRowNumber).setTripId(tripId).setShapeId(shapeId).build();
+    }
```

Comment:
Yes, there is no common utility containing these functions, and I also do not think it is necessary to group them to another place. Because even if the Gtfs type _XXX_ (e.g. GtfsTrip, GtfsShape, GtfsStopTime etc.) is fixed and the corresponding methods are all called "create_XXX_", the fields included for each "createXXX" method are very different according to the test contents. 
There are many fields that annotated as "ConditionallyRequired" rather than mandatory "Required", for example, shapeId for a trip. I added the field shapeId here, because it is very important for our tests for our specific rule, while when a trip is created in other tests for other rules, the fields' combinations have a pretty high possibility to be different. And, I don't think we should list all fields (that are not mandatory required) - there are too many.